fedimint-api-client/src/api/mod.rs

```diff
@@ -478,6 +481,14 @@ pub trait IGlobalFederationApi: IRawFederationApi {
 
     async fn await_transaction(&self, txid: TransactionId) -> TransactionId;
 
+    /// Waits for an output outcome to be available for a specific outpoint
+    #[deprecated(note = "https://github.com/fedimint/fedimint/issues/6671")]
+    async fn await_output_outcome(
+        &self,
+        outpoint: OutPoint,
+        decoders: &ModuleDecoderRegistry,
+    ) -> SerdeOutputOutcome;
```

Comment:
Why do we add this here? It seems the PR achieves its goal without ever using the fn (or I was to dumb to ctrl+f) by just defining a module-specific API.